spring-integration-core/src/main/java/org/springframework/integration/router/AbstractMappingMessageRouter.java

```diff
@@ -70,9 +68,8 @@
 	@ManagedAttribute
 	public void setChannelMappings(Map<String, String> channelMappings) {
 		Assert.notNull(channelMappings, "'channelMappings' must not be null");
-		Map<String, String> newChannelMappings = new ConcurrentHashMap<String, String>();
-		newChannelMappings.putAll(channelMappings);
-		this.doSetChannelMappings(newChannelMappings);
+		Map<String, String> newChannelMappings = new ConcurrentHashMap<String, String>(channelMappings);
+		doSetChannelMappings(newChannelMappings);
```

Comment:
`doSetChannelMappings` still does an in-place clear/replace on the field.

It's not clear why this uses it whereas the others replace the field directly. Perhaps for the logging?

If so, perhaps we should log all changes - mappings won't change frequently.